src/System.Memory/tests/TestHelpers.cs

```diff
@@ -30,6 +30,12 @@ public static void ValidateReferenceType<T>(this Span<T> span, params T[] expect
             AssertThrows<IndexOutOfRangeException, T>(span, (_span) => ignore = _span[expected.Length]);
         }
 
+        public static unsafe void ValidateNonNullEmpty<T>(this Span<T> span)
+        {
+            Assert.True(span.IsEmpty);
+            Assert.True(Unsafe.AsPointer(ref MemoryMarshal.GetReference(span)) != null);
```

Comment:
Nit: Assert.NotEqual would give a better message

Reply:
This is comparing pointers. `Assert.NotEqual` does not work for pointers. I would have to cast the pointers to IntPtr to make it work. It would make the code harder to understand.

Other places in the Span tests use `Assert.True/False` to compare pointers.